scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/extension/ExtensionMethods.scala

```diff
@@ -0,0 +1,32 @@
+package pl.touk.nussknacker.engine.extension
+
+import java.lang.reflect.Method
+
+object ExtensionMethods {
+
+  private val declarationsWithImplementations = Map[Class[_], Constructor](
+    classOf[Cast] -> new CastConstructor,
+  )
+
+  val registry: Set[Class[_]] = declarationsWithImplementations.keySet
+
+  def applies(clazz: Class[_]): Boolean = registry.contains(clazz)
+
+  def invoke(method: Method, target: Any, arguments: Array[Object], classLoader: ClassLoader): Any =
+    declarationsWithImplementations
+      .get(method.getDeclaringClass)
+      .map { case constructor: CastConstructor =>
```

Comment:
For now, instead of having sealed trait and pattern match on it, we could have this one:
```
def create(target: Any, classLoader: ClassLoader): Any 
```
in this interface.

Reply:
Done